src/apply.c

```diff
@@ -66,22 +66,24 @@ static int patch_image_init_fromstr(
 	if (git_pool_init(&out->pool, sizeof(git_diff_line)) < 0)
 		return -1;
 
-	for (start = in; start < in + in_len; start = end) {
-		end = memchr(start, '\n', in_len - (start - in));
+	if (in_len) {
```

Comment:
It would be easier to read if we just exited early in case `!in_len`

Reply:
Yay for the guard clause pattern! (Wasn't sure whether to add it here since it's not used consistently in the codebase, but I personally prefer it).